include/boost/astronomy/coordinate/sky_point.hpp

```diff
@@ -1,130 +1,157 @@
 #ifndef BOOST_ASTRONOMY_COORDINATE_SKY_POINT_HPP
 #define BOOST_ASTRONOMY_COORDINATE_SKY_POINT_HPP
 
-
 #include <string>
 #include <type_traits>
 #include <cmath>
 
-#include <boost/units/systems/angle/degrees.hpp>
 #include <boost/units/io.hpp>
 #include <boost/static_assert.hpp>
 
 #include <boost/astronomy/coordinate/frame.hpp>
 #include <boost/astronomy/detail/is_base_template_of.hpp>
 
+namespace boost { namespace astronomy { namespace coordinate {
+
+namespace bu = boost::units;
+namespace bg = boost::geometry;
 
-namespace boost
+/*!sky_point is used to represent a point(coordinate) in the sky*/
+template <typename CoordinateSystem>
+struct sky_point
 {
-    namespace astronomy
+    ///@cond INTERNAL
+    BOOST_STATIC_ASSERT_MSG((boost::astronomy::detail::is_base_frame_of
+            <boost::astronomy::coordinate::base_frame, CoordinateSystem>::value),
+            "Template argument is expected to be a frame class");
+    ///@endcond
+    
+protected:
+    CoordinateSystem point;
+
+public:
+    typedef CoordinateSystem system;
+    
+    //constructors  
+
+    //!default constructor
+    sky_point() {}
+
+    //!create point with the given coordinates
+    sky_point(CoordinateSystem const& object) : point(object) {}
+
+    //!create point with providing representation and differential class object
+    template <typename Representation, typename Differential>
+    sky_point
+    (
+        Representation const& representation_data,
+        Differential const& differential_data
+    )
+    {
+        BOOST_STATIC_ASSERT_MSG((boost::astronomy::detail::is_base_template_of
+            <boost::astronomy::coordinate::base_representation, Representation>::value),
+            "argument type is expected to be a representation class");
+
+        BOOST_STATIC_ASSERT_MSG((boost::astronomy::detail::is_base_template_of
+            <boost::astronomy::coordinate::base_differential, Differential>::value),
+            "argument type is expected to be a differential class");
+
+        point = CoordinateSystem(representation_data, differential_data);
```

Comment:
This statement will not be able to deduce the template parameter of `CoordinateSystem`. We need to pass the template parameter explicitly. Though I am not sure can you try making an object and see if it works. For other constructors too.

Reply:
I don't understand the problem here. And I have tried some examples and they ran fine

Reply:
Okay if they run fine then.